common/djangoapps/third_party_auth/decorators.py

```diff
@@ -30,3 +34,52 @@ def wrapped_view(request, *args, **kwargs):
         resp['X-Frame-Options'] = x_frame_option
         return resp
     return wraps(view_func, assigned=available_attrs(view_func))(wrapped_view)
+
+
+def tpa_hint_ends_existing_session(func):
+    """
+    Modifies a view function so that, if a tpa_hint URL parameter is received, the user is
+    already logged in, and the hinted SSO provider is so configured, the user is redirected
+    to a logout view and then back here, so that we can ensure that they're using the most
+    "fresh" information.
+    """
+
+    @wraps(func)
+    def inner(request, *args, **kwargs):
+        """
+        Check for a TPA hint in combination with a logged in user, and log the user out
+        if the hinted provider specifies that they should be.
+        """
+        sso_provider = None
+        provider_id = request.GET.get('tpa_hint')
+        logged_out = request.GET.get('logged_out') == 'yes'
```

Comment:
I see. I thought that the login flow might strip out the tpa_hint parameter after it gets used to hint the SSO login process, but it seems that that's not the case, so something like this is necessary. (An alternative could be to make the login flow strip out that parameter and then this view wouldn't need its own query param, but that's probably harder.)

However, the behavior you have just described is not what the two docstrings above say, and that's not what I would get from the variable name. So if you stick with this approach, please update the docstrings and rename this variable to something like `user_was_redirected_already` that better aligns with what you're checking. Since both @brittneyexline and I have already been unclear about this code, I think that being very explicit in the code itself is important ;)